frontend/src/scenes/persons/personsLogic.ts

```diff
@@ -54,6 +58,40 @@ export const personsLogic = kea<personsLogicType<PersonPaginatedResponse>>({
                 return match?.properties?.email || 'example@gmail.com'
             },
         ],
+        showSessions: [
+            (s) => [s.featureFlags],
+            (featureFlags: Record<string, string | boolean>): boolean => {
+                return !featureFlags[FEATURE_FLAGS.REMOVE_SESSIONS]
+            },
+        ],
+        showSessionRecordings: [
+            (s) => [s.featureFlags, s.currentTeam],
+            (featureFlags: Record<string, string | boolean>, currentTeam: TeamType): boolean => {
+                return !!featureFlags[FEATURE_FLAGS.REMOVE_SESSIONS] && currentTeam?.session_recording_opt_in
+            },
+        ],
+        showTabs: [
+            (s) => [s.showSessions, s.showSessionRecordings],
+            (showSessions: boolean, showSessionRecordings: boolean): boolean => {
+                return showSessions || showSessionRecordings
+            },
+        ],
+        currentTab: [
+            (s) => [s.activeTab, s.showSessionRecordings, s.showSessions],
+            (activeTab: PersonsTabType | null, showSessionRecordings: boolean, showSessions: boolean) => {
+                // Ensure the activeTab reflects a valid tab given the available tabs
+                if (!activeTab) {
+                    return showSessionRecordings ? PersonsTabType.SESSION_RECORDINGS : PersonsTabType.EVENTS
+                }
+                if (activeTab === PersonsTabType.SESSIONS && !showSessions) {
```

Comment:
Can this be combined with the above if check? Like `!activeTab || (activeTab === PersonsTabType.SESSIONS && !showSessions)`